src/ApolloHooksMutation.re

```diff
@@ -36,72 +30,98 @@ type controlledVariantResult('a) =
   | Error(error)
   | NoData;
 
-module Make = (Config: Config) => {
-  [@bs.module "graphql-tag"] external gql: ReasonApolloTypes.gql = "default";
-
-  type mutationResult = {. "data": option(Config.t)};
-
-  [@bs.deriving abstract]
-  type options = {
-    [@bs.optional]
-    variables: Js.Json.t,
-    [@bs.optional]
-    client: ApolloClient.generatedApolloClient,
-    [@bs.optional]
-    refetchQueries,
-    [@bs.optional]
-    awaitRefetchQueries: bool,
-    [@bs.optional]
-    update: (ApolloClient.generatedApolloClient, mutationResult) => unit,
-  };
+[@bs.module "graphql-tag"] external gql: ReasonApolloTypes.gql = "default";
+
+type mutationResult('a) = {. "data": 'a};
+
+[@bs.deriving abstract]
+type options('a) = {
+  [@bs.optional]
+  variables: Js.Json.t,
+  [@bs.optional]
+  mutation: option(ReasonApolloTypes.queryString),
+  [@bs.optional]
+  client: ApolloClient.generatedApolloClient,
+  [@bs.optional]
+  refetchQueries,
+  [@bs.optional]
+  awaitRefetchQueries: bool,
+  [@bs.optional]
+  update: (ApolloClient.generatedApolloClient, mutationResult('a)) => unit,
+};
 
-  type jsResult = {
-    .
-    "data": Js.Nullable.t(Js.Json.t),
-    "loading": bool,
-    "called": bool,
-    "error": Js.Nullable.t(error),
-  };
+type jsResult = {
+  .
+  "data": Js.Nullable.t(Js.Json.t),
+  "loading": bool,
+  "called": bool,
+  "error": Js.Nullable.t(error),
+};
 
-  type jsMutate = (. options) => Js.Promise.t(jsResult);
-  type mutation =
+type jsMutate('a) = (. options('a)) => Js.Promise.t(jsResult);
+type mutation('a, 'b) =
```

Comment:
This type is now not used, it can be included inside the type definition in `let useMutation =...` with some changes. I can fix that and push, if no one is against :)

Reply:
Sure.